In [8]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

print("Librerias importadas correctamente.")


Librerias importadas correctamente.


In [9]:
import urllib.request
import os

# PASO 1: Crear la carpeta si no existe (IMPORTANTE)
os.makedirs("../data/raw", exist_ok=True)
print("✓ Carpeta data/raw/ lista para usar")

# PASO 2: Descargar el archivo
url = "https://raw.githubusercontent.com/cajjster/data_files/main/vgsales.csv"
ruta_destino = "../data/raw/vgsales.csv"

# Intentar descargar el archivo. Si hay error, mostrarlo sin bloquear el programa
try:
    urllib.request.urlretrieve(url, ruta_destino)
    print("✓ El archivo se ha descargado correctamente en data/raw/vgsales.csv")
except Exception as e:
    print(f"❌ Error descargando: {e}")


✓ Carpeta data/raw/ lista para usar
✓ El archivo se ha descargado correctamente en data/raw/vgsales.csv


In [10]:
df = pd.read_csv(ruta_destino)
print("✓ Archivo cargado en DataFrame 'df' con éxito")

df.head()

✓ Archivo cargado en DataFrame 'df' con éxito


,Name,Platform,Year,Genre,Publisher,NA_Sales,EU_Sales,JP_Sales,Other_Sales,Global Sales
0,Wii Sports,Wii,2006,Sports,Nintendo,41.49,29.02,3.77,8.46,82.74
1,Super Mario Bros.,NES,1985,Platform,Nintendo,29.08,3.58,6.81,0.77,40.24
2,Mario Kart Wii,Wii,2008,Racing,Nintendo,15.85,12.88,3.79,3.31,35.82
3,Wii Sports Resort,Wii,2009,Sports,Nintendo,15.75,11.01,3.28,2.96,33.00
4,Pokemon Red/Pokemon Blue,GB,1996,Role-Playing,Nintendo,11.27,8.89,10.22,1.00,31.37


In [11]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 16291 entries, 0 to 16290
Data columns (total 10 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Name          16291 non-null  str    
 1   Platform      16291 non-null  str    
 2   Year          16291 non-null  int64  
 3   Genre         16291 non-null  str    
 4   Publisher     16291 non-null  str    
 5   NA_Sales      16291 non-null  float64
 6   EU_Sales      16291 non-null  float64
 7   JP_Sales      16291 non-null  float64
 8   Other_Sales   16291 non-null  float64
 9   Global Sales  16291 non-null  float64
dtypes: float64(5), int64(1), str(4)
memory usage: 1.2 MB


In [ ]:

#verificacion de plataformas, generos y publishers unicos

print("Las plataformas unicas: ",df["Platform"].nunique())
print("Los generos unicos: ",df["Genre"].nunique())
print("Publishers unicos: ",df["Publisher"].nunique())


Las plataformas unicas:  31
Los generos unicos:  12
Publishers unicos:  576


In [14]:
# visualizamos las plataformas unicas

print("Plaformas")
print(df["Platform"].unique())

# visualizamos los generos unicos

print("Generos")
print(df["Genre"].unique())

Plaformas
<StringArray>
[ 'Wii',  'NES',   'GB',   'DS', 'X360',  'PS3',  'PS2', 'SNES',  'GBA',
  '3DS',  'PS4',  'N64',   'PS',   'XB',   'PC', '2600',  'PSP', 'XOne',
   'GC', 'WiiU',  'GEN',   'DC',  'PSV',  'SAT',  'SCD',   'WS',   'NG',
 'TG16',  '3DO',   'GG', 'PCFX']
Length: 31, dtype: str
Generos
<StringArray>
[      'Sports',     'Platform',       'Racing', 'Role-Playing',
       'Puzzle',         'Misc',      'Shooter',   'Simulation',
       'Action',     'Fighting',    'Adventure',     'Strategy']
Length: 12, dtype: str


## 📋 Conclusiones de la Exploración

- El dataset contiene **16,291 videojuegos** con **10 variables** cada uno
- Cubre **40 años** de historia (1980 - 2020)
- **No hay valores nulos** — los datos están completos
- Hay **31 plataformas**, **12 géneros** y **576 publishers** diferentes
- Las ventas son muy desiguales: la mayoría vende poco (mediana: 0.17M) 
  pero pocos juegos venden muchísimo (máximo: 82.74M)
- El juego más vendido es **Wii Sports** con 82.74 millones de copias

### 🔜 Siguiente paso
Notebook 02 — Limpieza: corregir el nombre de la columna 
`Global Sales` (tiene espacio) y preparar los datos para el análisis.